plugins/chat/app/jobs/regular/chat/notify_mentioned.rb

```diff
@@ -145,13 +145,39 @@ def process_mentions(user_ids, mention_type)
         memberships = get_memberships(user_ids)
 
         memberships.each do |membership|
-          mention = ::Chat::Mention.find_by(user: membership.user, chat_message: @chat_message)
+          mention = find_mention(mention_type, membership.user.id)
           if mention.present?
             create_notification!(membership, mention, mention_type)
             send_notifications(membership, mention_type)
           end
         end
       end
+
+      def find_mention(mention_type, user_id)
+        mention_klass = resolve_mention_klass(mention_type)
+
+        target_id = nil
+        if mention_klass == ::Chat::UserMention
+          target_id = user_id
+        elsif mention_klass == ::Chat::GroupMention
+          target_id = Group.where("LOWER(name) = ?", mention_type).first.id
```

Comment:
We don't handle the cast where we don't find the group

Reply:
I think, the only thing we can do in that case is to write a message to logs a06deea9a810509951d26f66441202f591f56401? Is it what you had in mind? This won't handle the problem with the notification about the group mention, but at least other notifications for a chat message will be sent.

One note is that, it's highly unlikely a group won't be found here. The name of the group comes from `Notifier`, and `Notifier` makes sure that the group exists just a moment before this code runs. It's not really explicit, but `@parsed_mentions.group_mentions` in this place returns only existing groups:

https://github.com/discourse/discourse/blob/93ca13e5346aada027a7b9cadb43eba650889f20/plugins/chat/lib/chat/notifier.rb#L199-L200

That `ParsedMentions` class is a result of refactoring (we extracted that logic from Notifier), and it is still far from being ideal.

Something that I believe we can improve about the `NotifyMentioned` job is to make it work like this: the job can take the only parameter – message_id, and then it can load mentions for that message and send notifications for them. The current implementation when Notifier loads mentions, check them and then call the `NotifyMentioned` job with passing to it a hash with mentions feels too complex to me.